src/logic/taskcluster.ts

```diff
@@ -0,0 +1,67 @@
+// This file contains logic for the Taskcluster Third-Party Login
+
+import { UserCredentials } from '../types/types';
+
+export const prodTaskclusterUrl = 'https://firefox-ci-tc.services.mozilla.com';
+export const stagingTaskclusterUrl =
+  'https://stage.taskcluster.nonprod.cloudops.mozgcp.net';
+export const stagingClientId = 'perfcompare-staging-client';
+export const tcClientIdMap: Record<string, string> = {
+  'https://perf.compare': 'production',
+  'https://beta--mozilla-perfcompare.netlify.app': 'beta',
+  'http://localhost:3000': 'localhost-3000',
+};
+
+export const getTaskclusterParams = () => {
+  const redirectUri = `${window.location.origin}/taskcluster-auth`;
+  let tcParams = {
+    url: prodTaskclusterUrl,
+    redirectUri: redirectUri,
+    clientId: `perfcompare-${tcClientIdMap[window.location.origin]}-client`,
+  };
+  if (window.location.hash.includes('taskcluster-staging')) {
+    tcParams.url = stagingTaskclusterUrl;
+    tcParams.clientId = stagingClientId;
+  }
+  return tcParams;
+};
+
+const generateNonce = () => {
+  return window.crypto.randomUUID();
+};
+
+const getAuthCode = (tcParams: {
+  url: string;
+  redirectUri: string;
+  clientId: string;
+}) => {
+  const nonce = generateNonce();
+  sessionStorage.setItem('requestState', nonce); // The request state it's stored in sessionStorage so that it can be used in the callback
+  sessionStorage.setItem('taskclusterUrl', tcParams.url);
```

Comment:
nit: you can also add a comment here, such as
```
// The taskclusterUrl is also stored, so that the additional
// requests done by the callback endpoint use the same URL.
```